elasticapm/instrumentation/packages/elasticsearch.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import absolute_import
+
+import json
+import logging
+
+import elasticapm
+from elasticapm.instrumentation.packages.base import AbstractInstrumentedModule
+
+logger = logging.getLogger(__name__)
+
+
+API_METHOD_KEY_NAME = '__elastic_apm_api_method_name'
+BODY_REF_NAME = '__elastic_apm_body_ref'
+
+
+class ElasticsearchConnectionInstrumentation(AbstractInstrumentedModule):
+    name = 'elasticsearch_connection'
+
+    instrument_list = [
+        ("elasticsearch.connection.http_urllib3", "Urllib3HttpConnection.perform_request"),
+        ("elasticsearch.connection.http_requests", "RequestsHttpConnection.perform_request"),
+    ]
+
+    query_methods = (
+        'Elasticsearch.search',
+        'Elasticsearch.count',
+        'Elasticsearch.delete_by_query'
+    )
+
+    def call(self, module, method, wrapped, instance, args, kwargs):
+        args_len = len(args)
+        http_method = args[0] if args_len else kwargs.get('method')
+        http_path = args[1] if args_len > 1 else kwargs.get('url')
+        params = args[2] if args_len > 2 else kwargs.get('params')
+        body = params.pop(BODY_REF_NAME, None)
+
+        api_method = params.pop(API_METHOD_KEY_NAME, None)
+
+        signature = 'ES %s %s' % (http_method, http_path)
+        context = {'db': {'type': 'elasticsearch'}}
+        if api_method in self.query_methods:
+            query = []
+            # using both q AND body is allowed in some API endpoints / ES versions,
+            # but not in others. We simply capture both if they are there so the
+            # user can see it.
+            if 'q' in params:
+                # 'q' is already encoded to a byte string at this point
+                query.append('q=' + params['q'].decode('utf-8'))
+            if isinstance(body, dict) and 'query' in body:
+                query.append(json.dumps(body['query']))
+            context['db']['statement'] = '\n\n'.join(query)
+        if api_method == 'Elasticsearch.update':
+            if isinstance(body, dict) and 'script' in body:
+                context['db']['statement'] = json.dumps(body)
```

Comment:
Why doesn't this grab just the `script` here?  Seems care is taken to avoid capturing `doc` in the partial updated, but as is this would capture scripted upsert docs?

Reply:
yes, good catch!